waspc/src/Wasp/Generator/DbGenerator/Common.hs

```diff
@@ -1,38 +1,63 @@
 module Wasp.Generator.DbGenerator.Common
   ( dbMigrationsDirInDbRootDir,
+    serverPrismaClientOutputDirEnv,
+    webAppPrismaClientOutputDirEnv,
+    prismaClientOutputDirInModuleDir,
+    dbSchemaFileFromModuleDir,
     dbRootDirInProjectRootDir,
     dbSchemaChecksumOnLastDbConcurrenceFileProjectRootDir,
     dbSchemaChecksumOnLastGenerateFileProjectRootDir,
     dbSchemaFileInDbTemplatesDir,
-    dbSchemaFileInProjectRootDir,
     dbTemplatesDirInTemplatesDir,
     defaultMigrateArgs,
     getOnLastDbConcurrenceChecksumFileRefreshAction,
     MigrateArgs (..),
     RefreshOnLastDbConcurrenceChecksumFile (..),
+    DbSchemaChecksumOnLastDbConcurrenceFile,
+    DbSchemaChecksumOnLastGenerateFile,
+    PrismaDbSchema,
+    serverRootDirFromDbRootDir,
+    webAppRootDirFromDbRootDir,
+    dbSchemaFileInProjectRootDir,
+    prismaClientOutputDirEnvVar,
+    databaseUrlEnvVar,
+    DbSchemaChecksumFile,
   )
 where
 
 import StrongPath (Dir, File, File', Path', Rel, reldir, relfile, (</>))
 import qualified StrongPath as SP
 import Wasp.Common (DbMigrationsDir)
-import Wasp.Generator.Common (ProjectRootDir)
+import Wasp.Generator.Common (DbRootDir, ProjectRootDir, ServerRootDir, ModuleRootDir)
 import Wasp.Generator.Templates (TemplatesDir)
 
-data DbRootDir
-
 data DbTemplatesDir
 
--- | This file represents the checksum of schema.prisma at the point
+-- | This file represents the Prisma db schema.
+data PrismaDbSchema
+
+class DbSchemaChecksumFile f
+
+-- | This file represents the checksum of the Prisma db schema at the point
 -- at which we last interacted with the DB to ensure they matched.
 -- It is used to help warn the user of instances when they may need to migrate.
 data DbSchemaChecksumOnLastDbConcurrenceFile
 
--- | This file represents the checksum of schema.prisma
+instance DbSchemaChecksumFile DbSchemaChecksumOnLastDbConcurrenceFile
+
+-- | This file represents the checksum of the Prisma db schema
 -- at the point at which `prisma generate` was last run. It is used
 -- to know if we need to regenerate schema.prisma during web app generation or not.
 data DbSchemaChecksumOnLastGenerateFile
 
+instance DbSchemaChecksumFile DbSchemaChecksumOnLastGenerateFile
```

Comment:
We're using this to simulate union types in Haskell, check function `writeDbSchemaChecksumToFile` to see how it's used. The same goes for `ModuleRootDir` (better name pending).